### PR TITLE
735: integrate spline 0.5.x

### PR DESCRIPTION
**Pull request by benedeki**
* Using Spline 0.5.x in Enceladus (both in **SparkJobs** and **Menas**)

### CODE REVIEW

**Comment by Zejnilovic** [comment]
This PR will close #624
(if we don't reintroduce the issue)

File: pom.xml [hunk 1]
```diff
@@ -143,7 +143,7 @@
         <spark.hofs.version>0.4.0</spark.hofs.version>
         <spark.hats.version>0.2.1</spark.hats.version>
         <typesafe.config.version>1.3.1</typesafe.config.version>
-        <spline.version>0.3.9</spline.version>
+        <spline.version>0.4.1</spline.version>
```
Comment (Zejnilovic) [inline pom.xml]: Reminder before the final merge. Spline released 0.4.2. Might release more fixes by the time of merge.

**Comment by Zejnilovic** [comment]
Currently, if people click on the spline breadcrumbs they can open spline dashboard in any lineage window. Do we have a way to stop them?
image

**Comment by benedeki** [comment]
We ourselves don't have a way to prevent it.
But we can request the Spline guys to disable this and perhaps also the github link when the `embededMode` is set to true.
Created https://github.com/AbsaOSS/spline/issues/613

**Comment by Zejnilovic** [comment]
Tested with current develop of Spline 0.5.0. Seems to work fine. Both spark-jobs and Menas.

**Comment by Zejnilovic** [comment]
Spline link in the bottom left is not usable in embedded mode. Opened spline [issue 1272](https://github.com/AbsaOSS/enceladus/issues/1272)

File: menas/src/test/resources/application.properties [hunk 2]
```diff
@@ -67,7 +67,8 @@ menas.environment=localhost
 
 menas.spark.master=local[1]
 
-spline.urlTemplate=http://localhost:8080/spline/dataset/lineage/_search?path=%s&application_id=%s
+menas.lineage.readApiUrl=//localhost:8080/spline-gw/consumer
```
Comment (dk1844) [inline menas/src/test/resources/application.properties:70]: Is the leading `http` omitted intentionally (e.g. for supplying either `http` or `https`) or is this a mistake? If this is intentional, maybe there could be some comment explaining this?

Comment (Zejnilovic) [reply menas/src/test/resources/application.properties:70]: This is test properties. But in the main [properties](https://github.com/AbsaOSS/enceladus/pull/1175/files/2d10c5fbe907c46e69b27ad53d52614c4cd96440#diff-b9e082d316531342a516d81f8f47380cR75) there is a comment.

Comment (benedeki) [reply menas/src/test/resources/application.properties:70]: I will still add it the comment here as well, when moving to 0.5. 👍

File: pom.xml [hunk 3]
```diff
@@ -155,7 +155,6 @@
         <spark.hofs.version>0.4.0</spark.hofs.version>
         <spark.hats.version>0.2.1</spark.hats.version>
         <typesafe.config.version>1.3.1</typesafe.config.version>
-        <spline.version>0.4.2</spline.version>
```
Comment (Zejnilovic) [inline pom.xml]: Why not use `spline.ui.version` and `spline.agent.version`?

**Comment by Zejnilovic** [comment]
Closing as there is #1739 with newer spline

**Pull request closed without merging**